test/e2e/must_gather_operator_tests.go

```diff
@@ -1,114 +1,1186 @@
+						Verb:     "delete",
+						Group:    "",
+						Resource: "namespaces",
+					},
+				},
+			}
+			err := adminClient.Create(testCtx, sar)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sar.Status.Allowed).To(BeFalse(),
+				"Non-admin should NOT be able to delete namespaces")
+
+			// Verify non-admin cannot create ClusterRoles
+			sar2 := &authorizationv1.SubjectAccessReview{
+				Spec: authorizationv1.SubjectAccessReviewSpec{
+					User: nonAdminUser,
+					ResourceAttributes: &authorizationv1.ResourceAttributes{
+						Verb:     "create",
+						Group:    "rbac.authorization.k8s.io",
+						Resource: "clusterroles",
+					},
+				},
+			}
+			err = adminClient.Create(testCtx, sar2)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sar2.Status.Allowed).To(BeFalse(),
+				"Non-admin should NOT be able to create ClusterRoles")
+		})
+
+		ginkgo.It("non-admin user CANNOT perform admin operations using impersonation", func() {
+			ginkgo.By("Attempting to delete a namespace (should fail)")
+			testNS := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-delete-namespace",
+				},
+			}
+			err := nonAdminClient.Delete(testCtx, testNS)
+			Expect(err).To(HaveOccurred(), "Non-admin should NOT be able to delete namespaces")
+			Expect(apierrors.IsForbidden(err)).To(BeTrue(), "Should get Forbidden error")
+
+			ginkgo.By("Attempting to create a ClusterRole (should fail)")
+			testCR := &rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-unauthorized-clusterrole",
+				},
+			}
+			err = nonAdminClient.Create(testCtx, testCR)
+			Expect(err).To(HaveOccurred(), "Non-admin should NOT be able to create ClusterRoles")
+			Expect(apierrors.IsForbidden(err)).To(BeTrue(), "Should get Forbidden error")
+
+			ginkgo.GinkgoWriter.Println("✅ Non-admin user correctly blocked from admin operations")
+		})
+	})
+
+	ginkgo.Context("ServiceAccount Permissions", func() {
+		ginkgo.It("SA should have read access to cluster resources", func() {
+			saUser := fmt.Sprintf("system:serviceaccount:%s:%s", testNamespace, ServiceAccount)
+
+			checkSAPermission := func(verb, group, resource string) {
+				sar := &authorizationv1.SubjectAccessReview{
+					Spec: authorizationv1.SubjectAccessReviewSpec{
+						User: saUser,
+						ResourceAttributes: &authorizationv1.ResourceAttributes{
+							Verb:     verb,
+							Group:    group,
+							Resource: resource,
+						},
+					},
+				}
+				err := adminClient.Create(testCtx, sar)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(sar.Status.Allowed).To(BeTrue(),
+					fmt.Sprintf("SA should be able to %s %s.%s", verb, resource, group))
+			}
+
+			// Check read permissions for must-gather required resources
+			checkSAPermission("get", "", "nodes")
+			checkSAPermission("list", "", "nodes")
+			checkSAPermission("get", "", "pods")
+			checkSAPermission("list", "", "pods")
+			checkSAPermission("get", "", "namespaces")
+			checkSAPermission("list", "", "namespaces")
```

Comment:
are these SARC checks not redundant? In the same test we're adding the RBACs so it is bound to pass.